lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -760,10 +760,12 @@ def _on_set_password_action(self, event: ActionEvent):
         if self.upgrade_in_progress:
             event.fail("Setting password not supported while upgrade in-progress")
             return
-        if self.opensearch_peer_cm.deployment_desc().typ != DeploymentType.MAIN_ORCHESTRATOR:
-            event.fail("The action can be run only on the leader unit of the main cluster.")
+        if not self.opensearch_peer_cm.deployment_desc():
+            event.fail("The action can only be run once the deployment is complete.")
             return
-
+        if self.opensearch_peer_cm.deployment_desc().typ != DeploymentType.MAIN_ORCHESTRATOR:
+            event.fail("The action can be run only on the main cluster.")
```

Comment:
Sorry I missed these before!
```suggestion
            event.fail("The action can only be run on the main cluster.")
```